lxc-to-lxd/main_migrate.go

```diff
@@ -0,0 +1,602 @@
+	fmt.Println("Checking for unsupported LXC configuration keys")
+	keys := getUnsupportedKeys(getConfigKeys(conf))
+	for _, k := range keys {
+		if !strings.HasPrefix(k, "lxc.net.") &&
+			!strings.HasPrefix(k, "lxc.network.") &&
+			!strings.HasPrefix(k, "lxc.cgroup.") &&
+			!strings.HasPrefix(k, "lxc.cgroup2.") {
+			return fmt.Errorf("Found unsupported config key '%s'", k)
+		}
+	}
+
+	// Make sure we don't have a conflict
+	fmt.Println("Checking for existing containers")
+	containers, err := d.GetContainerNames()
+	if err != nil {
+		return err
+	}
+
+	found := false
+	for _, name := range containers {
+		if name == container.Name() {
+			found = true
+		}
+	}
+	if found {
+		return fmt.Errorf("Container already exists")
+	}
+
+	// Validate config
+	err = validateConfig(conf, container)
+	if err != nil {
+		return err
+	}
+
+	newConfig := make(map[string]string, 0)
+
+	value := getConfig(conf, "lxd.idmap")
+	if value == nil {
+		value = getConfig(conf, "lxd.id_map")
+	}
+	if value == nil {
+		// Privileged container
+		newConfig["security.privileged"] = "true"
+	} else {
+		// Unprivileged container
+		newConfig["security.privileged"] = "false"
+	}
+
+	newDevices := make(types.Devices, 0)
+
+	// Convert network configuration
+	err = convertNetworkConfig(container, newDevices)
+	if err != nil {
+		return err
+	}
+
+	// Convert storage configuration
+	err = convertStorageConfig(conf, newDevices)
+	if err != nil {
+		return err
+	}
+
+	// Convert environment
+	fmt.Println("Processing environment configuration")
+	value = getConfig(conf, "lxc.environment")
+	for _, env := range value {
+		entry := strings.Split(env, "=")
+		key := strings.TrimSpace(entry[0])
+		val := strings.TrimSpace(entry[len(entry)-1])
+		newConfig[fmt.Sprintf("environment.%s", key)] = val
+	}
+
+	// Convert auto-start
+	fmt.Println("Processing container boot configuration")
+	value = getConfig(conf, "lxc.start.auto")
+	if value != nil {
+		val, err := strconv.Atoi(value[0])
+		if err != nil {
+			return err
+		}
+
+		if val > 0 {
+			newConfig["boot.autostart"] = "true"
+		}
+	}
+
+	value = getConfig(conf, "lxc.start.delay")
+	if value != nil {
+		val, err := strconv.Atoi(value[0])
+		if err != nil {
+			return err
+		}
+
+		if val > 0 {
+			newConfig["boot.autostart.delay"] = value[0]
+		}
+	}
+
+	value = getConfig(conf, "lxc.start.order")
+	if value != nil {
+		val, err := strconv.Atoi(value[0])
+		if err != nil {
+			return err
+		}
+
+		if val > 0 {
+			newConfig["boot.autostart.priority"] = value[0]
+		}
+	}
+
+	// Convert apparmor
+	fmt.Println("Processing container apparmor configuration")
+	value = getConfig(conf, "lxc.apparmor.profile")
+	if value == nil {
+		value = getConfig(conf, "lxc.aa_profile")
+	}
+	if value != nil {
+		if value[0] == "lxc-container-default-with-nesting" {
+			newConfig["security.nesting"] = "true"
+		} else if value[0] != "lxc-container-default" {
+			newConfig["raw.lxc"] = fmt.Sprintf("lxc.aa_profile=%s\n", value[0])
```

Comment:
You should set the new key `lxc.apparmor.profile` as LXD contains code to translate from old to new not from new to old I think.